core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.io.Serializable;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Ordering;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class ViewMetadata implements Serializable {
+  static final int DEFAULT_VIEW_FORMAT_VERSION = 1;
+  static final int SUPPORTED_VIEW_FORMAT_VERSION = 1;
+
+  public static ImmutableViewMetadata.Builder buildFrom(ViewMetadata metadata) {
+    return ImmutableViewMetadata.builder()
+        .formatVersion(DEFAULT_VIEW_FORMAT_VERSION)
+        .location(metadata.location())
+        .properties(metadata.properties())
+        .currentVersionId(metadata.currentVersionId())
+        .versions(metadata.versions())
+        .schemas(metadata.schemas())
+        .currentSchemaId(metadata.currentSchemaId())
+        .history(metadata.history());
+  }
+
+  @Value.Check
+  void check() {
+    Preconditions.checkState(versions().size() > 0, "Expecting non-empty version log");
+    checkVersionLog(history());
+  }
+
+  public abstract int formatVersion();
+
+  public abstract String location();
+
+  public abstract Map<String, String> properties();
+
+  public abstract int currentVersionId();
+
+  public abstract List<ViewVersion> versions();
+
+  public abstract List<ViewHistoryEntry> history();
+
+  @Nullable
+  public abstract List<Schema> schemas();
+
+  @Nullable
+  public abstract Integer currentSchemaId();
+
+  public ViewVersion version(int versionId) {
+    return versionsById().get(versionId);
+  }
+
+  @Value.Derived
+  public ViewVersion currentVersion() {
+    return versionsById().get(currentVersionId());
+  }
+
+  @Value.Derived
+  public Map<Integer, ViewVersion> versionsById() {
+    return indexVersions(versions());
+  }
+
+  @Value.Derived
+  public Map<Integer, Schema> schemasById() {
+    return indexSchemas(schemas());
+  }
+
+  @Value.Derived
+  public Schema schema() {
+    return schemasById().get(currentSchemaId());
+  }
+
+  private static void checkVersionLog(List<ViewHistoryEntry> versionLog) {
+    Preconditions.checkState(versionLog.size() > 0, "Expecting non-empty version log");
+    Preconditions.checkState(
+        Ordering.from(Comparator.comparing(ViewHistoryEntry::timestampMillis))
+            .isOrdered(versionLog),
+        "Expected sorted version log entries.");
```

Comment:
I probably wouldn't bother with this, considering we can just sort the entries.

Reply:
Good point, there's not really any value to this check